hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.addressbook.impl.validators;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.FQDN_SIZE_TOO_LARGE;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.GOSSIP_ENDPOINT_CANNOT_HAVE_FQDN;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_ENDPOINT;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_GOSSIP_ENDPOINT;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_NODE_DESCRIPTION;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_SERVICE_ENDPOINT;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.IP_FQDN_CANNOT_BE_SET_FOR_SAME_ENDPOINT;
+import static com.hedera.node.app.spi.workflows.HandleException.validateFalse;
+
+import com.hedera.hapi.node.base.ServiceEndpoint;
+import com.hedera.node.config.data.NodesConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import javax.inject.Inject;
+
+public class AddressBookValidator {
+    /**
+     * Default constructor for injection.
+     */
+    @Inject
+    public AddressBookValidator() {
+        // Dagger2
+    }
+
+    /**
+     * Validates the node description.
+     *
+     * @param description The description to validate
+     * @param nodesConfig The nodes configuration
+     */
+    public void validateDescription(@Nullable final String description, @NonNull final NodesConfig nodesConfig) {
+        if (description == null || description.isEmpty()) {
+            return;
+        }
+        final var raw = description.getBytes(StandardCharsets.UTF_8);
+        final var maxUtf8Bytes = nodesConfig.nodeMaxDescriptionUtf8Bytes();
+        validateFalse(raw.length > maxUtf8Bytes, INVALID_NODE_DESCRIPTION);
+        validateFalse(containsZeroByte(raw), INVALID_NODE_DESCRIPTION);
+    }
+
+    private boolean containsZeroByte(@NonNull final byte[] bytes) {
+        boolean ret = false;
+        for (final byte b : bytes) {
+            if (b == 0) {
+                ret = true;
+                break;
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * Validates the gossip endpoint.
+     *
+     * @param endpointLst The list of GossipEndpoint to validate
+     * @param nodesConfig The nodes configuration
+     */
+    public void validateGossipEndpoint(
+            @Nullable final List<ServiceEndpoint> endpointLst, @NonNull final NodesConfig nodesConfig) {
+        validateFalse(endpointLst == null || endpointLst.isEmpty(), INVALID_GOSSIP_ENDPOINT);
+        validateFalse(endpointLst.size() > nodesConfig.maxGossipEndpoint(), INVALID_GOSSIP_ENDPOINT);
```

Comment:
Should this specify a different response code like `GOSSIP_ENDPOINTS_EXCEEDED_LIMIT`

Reply:
Changed to GOSSIP_ENDPOINTS_EXCEEDED_LIMIT.

Reply:
Will add SERVICE_ENDPOINTS_EXCEEDED_LIMIT in the next PR.